### DISCUSSION TITLE
Simple Docker Tutorials - A Discussion to Attract New Users

### DISCUSSION THREAD

**Discussion by tomtom215** [StarRocks/starrocks#13692] [Announcements]
**Idea to attract new user: simple ~5 minute quick start Docker tutorials**

Currently there is a docker image for starrocks:dev-env but it does not deliver the "quick deploy & try" experience that developers would want when trying out a new DB or tool.

When I tried this image, I was faced with an error where it cannot run on Apple's ARM processors. Then I tried it on an AMD ThreadRipper machine (no avx512) and it still took close to 30 minutes to finish running build.sh and once it completed, I somewhat expected that I would have a running starrocks DB with BE and FE components running in a docker container. 

Instead it seems I've just now compiled everything and the next steps to quickly get started loading data into the DB were unclear. 

For an ideal developer experience, I would wish for an officially maintained docker image that:

- Has all dependencies when the container is built (without the need to clone the git repo to the container, run the build.sh script, then pull thousands of java files from both mirrors to compile)
- Does not require any .CN domains/mirrors for build requirements (some corporate networks have geo-blocking with default rules to not allow .cn)
- Is up to date with the MAIN branch releases of starrocks
- Has clear get started documentation and a docker-compose example with networking best practices explained 
- Does not require any manually compiling/ initiating/ starting the DB after the container starts
- ARM compatible images that allow devs with Apple or other ARM processors to work without troubleshooting or manual compilation
- Accompanying officially maintained tutorials & docker-compose examples that walk through quick spin up of the DB and connecting to data sources to begin ingesting data
- - Wishfully including an up to date DBT-starrocks adapter for easy tutorials that allow developers to use the tools they already have available for data transformation

**Comment by alberttwong** [comment]
The new quickstart uses a docker image now.
